### DISCUSSION TITLE
Could not reach any node of database

### DISCUSSION THREAD

**Discussion by giancarlomelis** [ravendb/ravendb#16271] [Q&A] [answered]
Dear all,

I am having some difficulties accessing to database.
I have created an ASP.Net Core Web Api 2.0 service that allows inserting some documents.
I uploaded the certificate to the online server (SmarterASP).
I configured the application.json as follows:

```
  "Database": {
    "Urls": [
      "https://<my database>.ravendb.cloud"
    ],
    "DatabaseName": "<my database name>",
    "CertPath": "config/<my client certificate name>.pfx",
    "CertPass": "<certificate password>"
  }
```
  
Locally everything works fine, 
but when I deploy to the SmarterASP server and try to save a document, 
I get this error:

```
Could not reach any node of '<my database name>' database
None of 'my database name' database nodes are responding to the supervisor, the database is unreachable.. 
Things you may check: verify node is working, check for ports being blocked by firewall or similar software.

```
Could you give me a hand understanding what could be causing this issue?

Thanks!
Giancarlo.

**Comment by ayende** [comment]
The most typical reason for this is that you have limited the IPs that can reach the servers, so you are blocked at the network layer.

**Comment by giancarlomelis** [comment]
Thank you for your reply.
I added the IP address of the server where the WebApi service is running.
Now I am getting this error:

`Failed to retrieve database topology from all known nodes. https://<my database> -> The SSL connection could not be established, see inner exception. -> Authentication failed, see inner exception. -> The credentials supplied to the package were not recognized (The SSL connection could not be established, see inner exception.)`

**Reply by ayende** [reply]
When you run `curl https://your-db` what do you get?

**Reply by giancarlomelis** [reply]
If i try to get "https://a.free.__my_db_name._ravendb.cloud" from the SmarterASP server i obtain a  403 (Forbidden) status code

**Comment by giancarlomelis** [comment, marked as answer]
Bingo!
I need to pass the X509KeyStorageFlags parameter:
```
var store = new DocumentStore
{
    Urls = settings.Database.Urls,
    Database = settings.Database.DatabaseName,
    Certificate = new X509Certificate2(settings.Database.CertPath, settings.Database.CertPass, **X509KeyStorageFlags.MachineKeySet**)
};
```
